### DISCUSSION TITLE
Zibgee2MQTT device led bulb tuya TS0505B_1 no colors

### DISCUSSION THREAD

**Discussion by EyeVisionsNL** [Koenkk/zigbee2mqtt#24618] [Devices]
Hello everyone,

I first started with Zigbee2MQTT. So now I have an LED lamp that has colors, but I cannot control it. The LED lamp is recognized and supported. But if I change color xy, the example becomes black and the lamp does not change color. If I choose colorloop, you will see that it has all the colors. What am I doing wrong now?

**Comment by thokaro** [comment]
Hi,

i have exactly the same problem. Changing the brightness or Color Temp works, but changing color cx does not.

**Comment by wenzelt** [comment]
same here

**Comment by EyeVisionsNL** [comment]
I have found the solution. View here how to create external definition. 

Afterwards replace the code part with this.
    description: 'Zigbee 3.0 15W led light bulb E27 RGBCW',
    extend: tuya.extend.light_onoff_brightness_colortemp_color({colorTempRange: [153, 500], noConfigure: true}),
    configure: async (device, coordinatorEndpoint, logger) => {
        device.getEndpoint(1).saveClusterAttributeKeyValue('lightingColorCtrl', {colorCapabilities: 29});
    },
};


https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_1-pairing-the-device-with-zigbee2mqtt
